### PR TITLE
[receiver/awscloudwatch] Fix issue with log stream filtering in the cloudwatch receiver

### PR DESCRIPTION
**Pull request by rnishtala-sumo**
**Description:** 
Fix issue with log stream filtering in the cloudwatch receiver

**Link to tracking Issue:** 
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/22123

**Testing:**
Modified a unit test case that asserts on the number of requests being made for a single log stream prefix configured.

### CODE REVIEW

File: .chloggen/rnishtala_cw_receiver.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+change_type: bug_fix
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: receiversawscloudwatch
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: "Fixes a bug where the AWS CloudWatch receiver does the log stream filtering (using prefix) incorrectly."
+
+# Mandatory: One or more tracking issues related to the change. You can use the PR number here if no issue exists.
+issues: [https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/22123]
```
Comment (djaglowski) [inline .chloggen/rnishtala_cw_receiver.yaml:15]:
```suggestion
issues: [22123]
```
This should fix the chloggen failure

**Review by djaglowski — COMMENTED** [review]
LGTM.

@schmikei, will you also take a look?

File: .chloggen/rnishtala_cw_receiver.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Use this changelog template to create an entry for release notes.
+# If your change doesn't affect end users, such as a test fix or a tooling change,
+# you should instead start your pull request title with [chore] or use the "Skip Changelog" label.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: bug_fix
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: receiversawscloudwatch
```
Comment (schmikei) [inline .chloggen/rnishtala_cw_receiver.yaml:9]:
```suggestion
component: awscloudwatchreceiver
```

I think this should be formatted like this right?

**Review by schmikei — APPROVED** [review]

File: .chloggen/rnishtala_cw_receiver.yaml (hunk 2 above)
Comment (rnishtala-sumo) [reply .chloggen/rnishtala_cw_receiver.yaml:9]: oh I meant, `receivers/awscloudwatch`

File: .chloggen/rnishtala_cw_receiver.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# Use this changelog template to create an entry for release notes.
+# If your change doesn't affect end users, such as a test fix or a tooling change,
+# you should instead start your pull request title with [chore] or use the "Skip Changelog" label.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: bug_fix
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: receivers/awscloudwatch
```
Comment (djaglowski) [inline .chloggen/rnishtala_cw_receiver.yaml:9]:
```suggestion
component: awscloudwatchreceiver
```
We could be more consistent about this but in the case of changelog entries, we use this format. (See example in line above)

**Review by djaglowski — APPROVED** [review]

**Pull request merged**